lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -215,6 +260,49 @@ def test__has_access_descriptor_when_not_in_preview_mode(self):
         # Start date in the future.
         mock_unit.start = datetime.datetime.now(pytz.utc) + datetime.timedelta(days=1)  # release date in the future
         self.verify_access(mock_unit, False)
+        self.verify_error_type(mock_unit, access_response.StartDateError)
+
+    @ddt.data(
+        (TOMORROW, ADVERTISED_START, ADVERTISED_START),
+        (TOMORROW, None, defaultfilters.date(TOMORROW, "DATE_FORMAT")),
+        (DEFAULT_START_DATE, ADVERTISED_START, ADVERTISED_START),
+        (DEFAULT_START_DATE, None, None)
+    )
+    @ddt.unpack
+    @patch.dict('django.conf.settings.FEATURES', {'DISABLE_START_DATES': False})
+    def test__has_access_descriptor_start_message(self, start, advertised_start, expected_message):
+        """
+        Tests that the correct start message is returned in the case the course has not started
+        Cases: start date not default and advertised start set, start date default and advertised start set
+               start date not default and no advertised start, Start date default and no advertised start
```

Comment:
No reason to enumerate the cases in the docstring, since they are nicely readable in the ddt declaration.  Otherwise, it adds docstring maintenance issues.